File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        arrsum,d=0,k+k+1
        ans = [-1 for i in range(len(nums))]
        for i in range(len(nums)):
            arrsum+=nums[i]
            if i+1>=d:
                if i>=d:
                    arrsum-=nums[i-d]
                ans[i-k] = (arrsum//d)
        return ans

File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int64_t sum = 0,d = k+k+1;
        vector<int>ans(nums.size(),-1);
        for(int i=0;i<nums.size();++i){
            sum+=nums[i];
            if(i+1>=d){
                if(i>=d){
                    sum-=nums[i-d];
                }
                ans[i-k]=sum/d;
            }
        }
        return ans;
    }
};